src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -101,6 +105,19 @@ def auth(self, url: str, cluster_alias: str, username: str,
 
         return False
 
+    def get_cors_endpoints_string(self, hub_url):
+        parsed_url = urlparse(hub_url)
+        hostname = parsed_url.hostname
+        cors_endpoints_list = []
+        orch = OrchClient.instance()
+        inventory_hosts = [host.to_json() for host in orch.hosts.list()]
+        for host in inventory_hosts:
+            host_addr = host['addr']
+            host_url = hub_url.replace(hostname, host_addr)
```

Comment:
Can this create issue if the user is operating over a hostname and the cors has ip addresses set. Can you check that usecase?

Reply:
Thanks @cloudbehl for pointing out, To cover this scenario we need to set the cross origin url with IP url and the fqdn as well. updated the PR accordingly.